vision_ocv/benchmark/modules/src/BM_ocv_vision_factory_test.cpp

```diff
@@ -0,0 +1,175 @@
+    std::atomic<int> eventsListened; // To be removed when pub/sub params will be implemented
+    std::function<void(kpsr::vision_ocv::ImageData &)> buildFunction;
+
+private:
+    std::unique_ptr<kpsr::vision_ocv::ImageDataFactory> _imageDataFactory;
+    std::function<void(const kpsr::vision_ocv::ImageData &, kpsr::vision_ocv::ImageData &)>
+        _cloneFunction;
+    std::unique_ptr<kpsr::EventEmitterMiddlewareProvider<kpsr::vision_ocv::ImageData>>
+        _eventEmitterProvider;
+    std::unique_ptr<kpsr::high_performance::EventLoopMiddlewareProvider<16>> _eventLoopProvider;
+};
+
+BENCHMARK_DEFINE_F(FactoryPerformanceTest, SimpleTest)(benchmark::State &state)
+{
+    bool useEventLoop = state.range(1);
+    setListener(state);
+
+    int eventsPublished = 0;
+
+    kpsr::vision_ocv::ImageData event;
+    buildFunction(event);
+    for (auto _ : state) {
+        event.timestamp = std::chrono::duration_cast<std::chrono::milliseconds>(
+                              std::chrono::system_clock::now().time_since_epoch())
+                              .count();
+        publisher->publish(event);
+        eventsPublished++;
+        while (useEventLoop && eventsListened.load() < eventsPublished) {
+            // TODO: Change loop condition to use events parameters from pub/sub objects
+            // Same with the other functions
+            std::this_thread::sleep_for(std::chrono::microseconds(1));
+        }
+    }
+
+    state.counters["publishedEvents"] = eventsPublished;
+    state.counters["listenerEvents"] = eventsListened.load();
+}
+
+BENCHMARK_DEFINE_F(FactoryPerformanceTest, ReadingFromDiskTest)(benchmark::State &state)
+{
+    bool useEventLoop = state.range(1);
+    setListener(state);
+
+    int eventsPublished = 0;
+
+    kpsr::vision_ocv::FileImageStreamingService streamService(nullptr,
+                                                              publisher,
+                                                              TEST_DATA,
+                                                              true,
+                                                              width,
+                                                              height,
+                                                              "raw_image",
+                                                              "serviceStream");
+    streamService.startup();
+
+    for (auto _ : state) {
+        streamService.runOnce();
+        eventsPublished++;
+        while (useEventLoop && eventsListened.load() < eventsPublished) {
+            std::this_thread::sleep_for(std::chrono::microseconds(1));
+        }
+    }
+
+    state.counters["publishedEvents"] = eventsPublished;
+    state.counters["listenerEvents"] = eventsListened.load();
+
+    streamService.shutdown();
+}
+
+BENCHMARK_REGISTER_F(FactoryPerformanceTest, SimpleTest)
+    ->UseManualTime()
+    ->Unit(benchmark::kMillisecond)
+    ->ArgsProduct({{0, 4}, {false, true}, {10, 50, 100, 1000, 10000}});
```

Comment:
also add this to make arguments readable in output:
```suggestion
    ->ArgsProduct({{0, 4}, {false, true}, {10, 50, 100, 1000, 10000}}) 
    ->ArgNames({PoolSize, WithEventLoop, ImageSize});
```